### ISSUE TITLE
Question about "ignore_index": 19

### ISSUE THREAD

**Issue by zllrunning** [donnyyou/torchcv#6]
In hypes/seg/cityscape/*_cityscape_seg.json, 
```"ignore_index": 19```
But in datasets/seg/fs_data_loader.py
```encoded_labelmap = np.ones(shape=(shape[0], shape[1]), dtype=np.float32) * 255```
so labels treated as void will be equel to 255.
My question is that should ignore_index be equel to 255?
Looking forward to your reply.

**Comment by youdonny** [comment]
https://github.com/CVBox/PyTorchCV/blob/5191d0ddc5c42a4cc8dc5451aa14c263c2f3e77f/datasets/seg_data_loader.py#L53

**Comment by zllrunning** [comment]
Thank you very much!

**Comment by zllrunning** [comment]
A new question...
We only use 19 classes excluding the void category and rare cases. And the void category and rare cases will be mapped to 255(at last 19).  The last Conv layer outputs 19 channels. When training,  we will use "ignore_index" to ignore 19.
But when testing, we will use np.argmax to get the label of the pixel.  If the label of the pixel in ground truth is void, then the label obtained when testing is untrustworthy, equivalent to a random label, because this part is not considered during training.
Is this the meaning?

Thank you very much in advance for your help!

**Comment by youansheng** [comment]
Yep, you're right! Ignored area is not considered during training, so the testing result is random.
